daffodil-core/src/main/scala/org/apache/daffodil/dsom/ModelGroup.scala

```diff
@@ -160,7 +160,10 @@ abstract class ModelGroup(index: Int)
       groupMembers.forall { _.hasStaticallyRequiredOccurrencesInDataRepresentation }
   }
 
-  def groupMembers: Seq[Term]
+  final lazy val groupMembers: Seq[Term] =
+    schemaSet.sharedGroupMembersFactory.getShared(shareKey, groupMembers1)
```

Comment:
It might be worth adding a comment around some of these parts, considering this is part of the "magic". For example, it seems important to point out that groupMembers1 (any chance a better name could be used) is a lazy val and that the ``getShared`` value parameter is passed by reference. If either of those changes I think this would still end up allocating some unnecessary things.

Reply:
I will add comments.

I will add documentation.

public groupMembers, 
protected groupMembersDef, 
and on the objects which aren't Terms, 
public groupMembersNotShared.